src/sim_engine/pool.rs

```diff
@@ -104,6 +114,19 @@ impl Pool for SimPool {
         Ok(())
     }
 
+    fn create_snapshot(&mut self, snapshot_name: &str, source: &str) -> EngineResult<()> {
+
+        let parent_id = try!(self.check_snapshot_source(source));
+
+        try!(self.create_filesystem(&snapshot_name, &String::from(""), None));
```

Comment:
The pool can't insert it into the BTreeMap and return a mutable reference, right?

Are you suggesting we return the mutable object, then the caller needs to make another call to get it inserted into the BTreeMap once they are done with it?